api/src/main/java/io/opentelemetry/metrics/LabelSet.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.metrics;
+
+/**
+ * LabelSet is an implementation-level interface that represents a set of Labels for use as
+ * pre-defined labels in the metrics API.
+ */
+public interface LabelSet {}
```

Comment:
An interface with no methods on it is a pretty bad smell. Why does this need to be an interface, rather than a concrete implementation?

Reply:
It is a good valid point, but this is how it is defined in the specs. The main idea is to allow different implementation to do any encoding that is more appropriate for them.

Reply:
Ugh. Well, if there are invariants that have to hold about implementations, can we add that to the javadoc? For instance, there should be a requirement that LabelSets that have the same key/value pairs should have the same equals/hashcode, yes?

Reply:
Done.